src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -1010,9 +1011,9 @@ public static boolean onGuiCharTypedPost(Screen guiScreen, char codePoint, int m
         return MinecraftForge.EVENT_BUS.post(event);
     }
 
-    public static void onRecipesUpdated()
+    public static void onRecipesUpdated(List<IRecipe<?>> recipes)
     {
-        Event event = new RecipesUpdatedEvent();
+        Event event = new RecipesUpdatedEvent(recipes);
```

Comment:
This is a binary breaking change, and also you should probably pass in the recipe book/recipe manager and not the raw list of recipes... As the manager has more information. And should have access to this list right?

Reply:
The manager does not have access to the passed list only the total recipes available on the client.  And while technically breaking, unless someone is manually firing the event (which nobody should ever be, as recipe loading is internal) this will not be a problem.

It could stand to access the recipe manager, so that can be added, but both may want to be present.

Reply:
Is this recipe list just a subset? How does this system work?
Can the server send remove, as well as add?
It's been a while since I've looked at recipes, but arnt they a full Forge registry and thus synced at connect? At-least ID wise.

Reply:
We removed the recipe registry when the manager was introduced in 1.13, so recipes no longer have ids and can be synced whenever desired.  However it looks like the vanilla expectancy is that the packet always sends the entire recipe list to the client, so the collection can be dropped in favor of the entire manager.

As for when the packet is sent, it is sent on login and also re-sent whenever /reload is executed.